src/evd.cpp

```diff
@@ -140,8 +140,10 @@ void evd_classic_tol(struct matrix_t Xmat, struct vector_t evec, struct matrix_t
 
         double A_ip, A_iq;
         for (int i = 0; i < n; ++i) {
-            Q[n * i + p] = c * Q[n * i + p] - s * Q[n * i + q];
-            Q[n * i + q] = s * Q[n * i + p] + c * Q[n * i + q];
+            double Q_ip = Q[n * i + p];
```

Comment:
Why is this stored in a variable before updating?

Reply:
Because otherwise the update of Q[n*i+p] has an effect on the second line, which should not be the case because they happen at the same time.